SimpleKeychainTests/SimpleKeychainSpec.swift

```diff
@@ -1,405 +1,373 @@
+    }
+    
+    #if canImport(LocalAuthentication) && !os(tvOS)
+    func testInitializationWithCustomLocalAuthenticationContext() {
+        let context = LAContext()
+        sut = SimpleKeychain(context: context)
+        XCTAssertEqual(sut.context, context)
+    }
+    #endif
+    
+    func testStoringStringItemUnderNewKey() {
+        let key = UUID().uuidString
+        XCTAssertNoThrow(try sut.set("foo", forKey: key))
+    }
+    
+    func testStoringStringItemUnderExistingKey() {
+        let key = UUID().uuidString
+        try? sut.set("foo", forKey: key)
+        XCTAssertNoThrow(try sut.set("bar", forKey: key))
+    }
+    
+    func testStoringDataItemUnderNewKey() {
+        let key = UUID().uuidString
+        XCTAssertNoThrow(try sut.set(Data(), forKey: key))
+    }
+    
+    func testStoringDataItemUnderExistingKey() {
+        let key = UUID().uuidString
+        try? sut.set(Data(), forKey: key)
+        XCTAssertNoThrow(try sut.set(Data(), forKey: key))
+    }
+    
+    func testDeletingItem() {
+        let key = UUID().uuidString
+        try? sut.set("foo", forKey: key)
+        XCTAssertNoThrow(try sut.deleteItem(forKey: key))
+    }
+    
+    func testDeletingNonExistingItem() {
+        XCTAssertThrowsError(try sut.deleteItem(forKey: "SHOULDNOTEXIST"))
+    }
+    
+    func testDeletingAllItems() {
+        let key = UUID().uuidString
+        try? sut.set("foo", forKey: key)
+        try? sut.deleteAll()
+        XCTAssertThrowsError(try sut.string(forKey: key))
+    }
+    
+    #if os(macOS)
+    func testIncludingLimitAllAttributeWhenDeletingAllItems() {
+        var limit: String?
+        sut.remove = { query in
+            let key = kSecMatchLimit as String
+            limit = (query as NSDictionary).value(forKey: key) as? String
+            return errSecSuccess
+        }
+        try? sut.deleteAll()
+        XCTAssertEqual(limit, kSecMatchLimitAll as String)
+    }
+    #else
+    func testNotIncludingLimitAllAttributeWhenDeletingAllItems() {
+        var limit: String? = ""
+        sut.remove = { query in
+            let key = kSecMatchLimit as String
+            limit = (query as NSDictionary).value(forKey: key) as? String
+            return errSecSuccess
+        }
+        try? sut.deleteAll()
+        XCTAssertNil(limit)
+    }
+    #endif
+    
+    func testRetrievingStringItem() {
+        let key = UUID().uuidString
+        try? sut.set("foo", forKey: key)
+        XCTAssertEqual(try? sut.string(forKey: key), "foo")
+    }
+    
+    func testRetrievingDataItem() {
+        let key = UUID().uuidString
+        try? sut.set("foo", forKey: key)
+        XCTAssertNotNil(try? sut.data(forKey: key))
+    }
+    
+    func testRetrievingNonExistingStringItem() {
+        XCTAssertThrowsError(try sut.string(forKey: "SHOULDNOTEXIST"))
+    }
+    
+    func testRetrievingNonExistingDataItem() {
+        XCTAssertThrowsError(try sut.data(forKey: "SHOULDNOTEXIST"))
```

Comment:
Is there any way to assert that the error thrown is `SimpleKeychainError.itemNotFound`?

Reply:
Yes, updated it to verify if the error thrown is `SimpleKeychainError.itemNotFound`, please take a look again